kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/dhcp/DhcpServerConfigIP4.java

```diff
@@ -51,12 +56,117 @@
      * @param dnsServers
      *            the DNS servers that will get passed to DHCP clients if passDns is true
      */
+	@Deprecated
     public DhcpServerConfigIP4(String interfaceName, boolean enabled, IP4Address subnet, IP4Address routerAddress,
             IP4Address subnetMask, int defaultLeaseTime, int maximumLeaseTime, short prefix, IP4Address rangeStart,
             IP4Address rangeEnd, boolean passDns, List<IP4Address> dnsServers) {
 
-        super(interfaceName, enabled, subnet, routerAddress, subnetMask, defaultLeaseTime, maximumLeaseTime, prefix,
+    	super(interfaceName, enabled, subnet, routerAddress, subnetMask, defaultLeaseTime, maximumLeaseTime, prefix,
                 rangeStart, rangeEnd, passDns, dnsServers);
+    }
+	
+	/**
+     * Private constructor for a DhcpServerConfigIP4 to be invoked by the newDhcpServerConfigIP4 method
+     *
+     * @param enabled
+     *            the status of the DhcpServer as a boolean
+     * @param interfaceName
+     *            the interface name associated with the DhcpServerConfig
+     * @param subnet
+     *            the subnet of the DhcpServerConfig
+     * @param routerAddress
+     *            the router IPAddress
+     * @param subnetMask
+     *            the subnet mask of the DhcpServerConfig
+     * @param defaultLeaseTime
+     *            the default lease time to issue to DHCP clients
+     * @param maximumLeaseTime
+     *            the maximum lease time to issue to DHCP clients
+     * @param prefix
+     *            the network prefix associated with the DhcpServerConfig
+     * @param rangeStart
+     *            the network starting address to issue to DHCP clients
+     * @param rangeEnd
+     *            the network ending address to issue to DHCP clients
+     * @param passDns
+     *            whether or not to pass DNS to DHCP clients
+     * @param dnsServers
+     *            the DNS servers that will get passed to DHCP clients if passDns is true
+     */
+	private DhcpServerConfigIP4(boolean enabled, String interfaceName, IP4Address subnet, IP4Address routerAddress,
```

Comment:
Since we are deprecating the old, why don't we try to have a more cleaner approach by passing an object, instead of having so many arguments? In this way we should be able to better manage errors (checks can be performed in advance in the object), and increase readability. What do you think?